pkg/cmd/extension/push/cmd_test.go

```diff
@@ -15,122 +15,84 @@
-				By("changing to a non-workspace dir")
-				dir := chdir("testdata")
-
-				By("running command")
-				c.SetArgs([]string{"extension", "push", localRegistryWasmImageRef})
-				err := cmdutil.Execute(c)
-				Expect(err).To(HaveOccurred())
-
-				By("verifying command output")
-				Expect(stdout.String()).To(BeEmpty())
-				Expect(stderr.String()).To(Equal(fmt.Sprintf(`Error: there is no extension directory at or above: %s
-
-Run 'getenvoy extension push --help' for usage.
-`, dir)))
-			})
-		})
-	})
-})
+// relativeWorkspaceDir points to a usable pre-initialized workspace
+const relativeWorkspaceDir = "testdata/workspace"
+
+// localRegistryWasmImageRef corresponds to a Docker container running the image "registry:2"
+// As this is not intended to be an end-to-end test, this could be improved to use a mock/fake HTTP registry instead.
+const localRegistryWasmImageRef = "localhost:5000/getenvoy/sample"
+
+// When unspecified, we default the tag to Docker's default "latest". Note: recent tools enforce qualifying this!
+const defaultTag = "latest"
+
+// TestGetEnvoyExtensionPush shows current directory is usable, provided it is a valid workspace.
+func TestGetEnvoyExtensionPush(t *testing.T) {
+	_, revertWd := extension.RequireChDir(t, relativeWorkspaceDir)
+	defer revertWd()
+
+	// Run "getenvoy extension push localhost:5000/getenvoy/sample"
+	cmd, stdout, stderr := extension.NewRootCommand()
+	cmd.SetArgs([]string{"extension", "push", localRegistryWasmImageRef})
+	err := cmdutil.Execute(cmd)
+
+	// A fully qualified image ref includes the tag
+	imageRef := localRegistryWasmImageRef + ":" + defaultTag
+
+	// Verify stdout shows the latest tag and the correct image ref
+	require.NoError(t, err, `expected no error running [%v]`, cmd)
+
+	require.Contains(t, stdout.String(), fmt.Sprintf(`Using default tag: %s
+Pushed %s
+digest: sha256`, defaultTag, imageRef), `unexpected stderr after running [%v]`, cmd)
+	require.Empty(t, stderr.String(), `expected no stderr running [%v]`, cmd)
+}
+
+func TestGetEnvoyExtensionPushFailsOutsideWorkspaceDirectory(t *testing.T) {
+	// Change to a non-workspace dir
+	dir, revertWd := extension.RequireChDir(t, relativeWorkspaceDir+"/..")
+	defer revertWd()
+
+	// Run "getenvoy extension push localhost:5000/getenvoy/sample"
+	cmd, stdout, stderr := extension.NewRootCommand()
+	cmd.SetArgs([]string{"extension", "push", localRegistryWasmImageRef})
+	err := cmdutil.Execute(cmd)
+
+	// Verify the command failed with the expected error
+	expectedErr := "there is no extension directory at or above: " + dir
+	require.EqualError(t, err, expectedErr, `expected an error running [%v]`, cmd)
+	require.Empty(t, stdout.String(), `expected no stdout running [%v]`, cmd)
+	expectedStderr := fmt.Sprintf("Error: %s\n\nRun 'getenvoy extension push --help' for usage.\n", expectedErr)
+	require.Equal(t, expectedStderr, stderr.String(), `expected stderr running [%v]`, cmd)
+}
+
+// TestGetEnvoyExtensionPushWithExplicitFileOption shows
```

Comment:
looks like the comment is incomplete :smile:

Reply:
someone died while writing